voxel51/apps/auth.py

```diff
@@ -0,0 +1,91 @@
+'''
+Application authentication module for the Voxel51 Vision Services API.
+
+| Copyright 2017-2019, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+'''
+import logging
+import os
+
+from voxel51.auth import Token
+import voxel51.utils as voxu
+
+
+logger = logging.getLogger(__name__)
+
+
+TOKEN_ENVIRON_VAR = "VOXEL51_APP_TOKEN"
+KEY_HEADER = "x-voxel51-application"
+USER_HEADER = "x-voxel51-application-user"
+TOKEN_PATH = os.path.join(
+    os.path.expanduser("~"), ".voxel51", "app-token.json")
+
+
+def activate_application_token(path):
+    '''Activates the application token by copying it to
+    ``~/.voxel51/app-token.json``.
+
+    Subsequent API instances created will now use this token for
+    authentication.
+
+    Args:
+        path (str): the path to an :class:`ApplicationToken` JSON file
+    '''
+    voxu.copy_file(path, TOKEN_PATH)
+    logger.info("ApplicationToken successfully activated")
+
+
+def deactivate_application_token():
+    '''Deactivates (deletes) the currently active application token, if any.
+
+    The active application token is at ``~/.voxel51/app-token.json``.
```

Comment:
Looks like double ` marks here?

Reply:
Required by the autodoc tool we're using here.